astropy/tests/runner.py

```diff
@@ -109,6 +110,22 @@ def __new__(cls, *args, **kwargs):
         cls.keywords = OrderedDict()
         doc_keywords = ""
         for name, func in sorted_keywords:
+            # Here we test if the function has been overloaded to return
+            # NotImplemented which is the way to disable arguments on
+            # subclasses. If it has been disabled we need to remove it from the
+            # default keywords dict. We do it in the try except block because
+            # we do not have access to an instance of the class, so this is
+            # going to error unless the method is just doing `return
+            # NotImplemented`.
+            try:
+                # Second argument is False, as it is normally a bool.
+                # The other two are placeholders for objects.
+                if func(None, False, None) is NotImplemented:
```

Comment:
Is the first `None` here a substitute for a `self`? (I'm guessing this as below `func` has only two arguments.) If so, wouldn't this always fail because `None` has the wrong class?